tests/test_backends.py

```diff
@@ -0,0 +1,164 @@
+        strategy = Mock()
+        self.cas_backend = CASAuth(strategy=strategy)
+
+    def test_cas_service_url_defined_in_settings(self):
+        """
+        This method is used to verify the CAS_SERVICE_URL variable in settings is defined
+        Expected behavior:
+        CASAuth.auth_url includes a query param called 'service' which value is a parsed string with the CAS_SERVICE_URL
+        """
+        auth_url = self.cas_backend.auth_url()
+        self.assertIn(urllib.parse.quote(settings.CAS_SERVICE_URL), auth_url)
+
+    @override_settings(
+        CAS_SERVER_URL=None
+    )
+    def test_cas_server_url_undefined_in_settings(self):
+        """
+        This method is used to verify the CAS_SERVER_URL variable in settings is defined
+        Expected behavior:
+        raise AuthMissingParameter
+        """
+        self.assertRaises(AuthMissingParameter, self.cas_backend.auth_url)
+
+    @override_settings(
+        CAS_SERVICE_URL=None
+    )
+    def test_cas_service_url_undefined_in_settings(self):
+        """
+        This method is used to verify the CAS_SERVICE_URL variable in settings is defined
+        Expected behavior:
+        raise AuthMissingParameter
+        """
+        self.assertRaises(AuthMissingParameter, self.cas_backend.auth_url)
+
+    @patch('openedx_cas.backends.CASAuth.cas_validation')
+    def test_auth_complete_ticket_validated(self, cas_validation):
+        """
+        This method is used to verify that the flow is valid when the ticket is valid
+        Expected behavior:
+        Response returned by auth_complete should include the correspoding values when the ticket is valid
+        """
+        cas_validation.return_value = {
+            'username': 'edxplatform'
+        }
+        service_response = {"ticket": "213123123-123123123"}
+        request = Mock(GET=service_response)
+        self.cas_backend.auth_complete(request=request)
+
+        self.cas_backend.strategy.authenticate.assert_called_with(request=request, response={'username': 'edxplatform'},
+                                                                  backend=self.cas_backend)
+
+    @unpack
+    @data({'username': 'test_has_value',
+           'email': 'test_has_value',
+           'fullname': 'test_has_value',
+           'first_name': 'test_has_value',
+           'last_name': 'test_has_value'},
+          {'username': '',
+           'first_name': '',
+           'last_name': ''},
+          {'username': ''},
+          )
+    def test_get_user_details(self, **kwargs):
+        """
+        This method is used to verify that the user_details are returned correctly whenever those are or not in the
+        response
+        Expected behavior:
+        Returns a dict with the corresponding information
+        """
+        response = self.cas_backend.get_user_details(kwargs)
+
+        for key, value in kwargs.items():
+            if value:
+                self.assertEqual(value, response[key])
```

Comment:
self.assertDictEquals

Reply:
That's not the same. If some key is not defined (mail, full_name, first_name, etc) it will not check it. The test still works.